pkg/table/tables/tables.go

```diff
@@ -300,26 +315,26 @@ func (t *TableCommon) getCols(mode getColsMode) []*table.Column {
 
 // Cols implements table.Table Cols interface.
 func (t *TableCommon) Cols() []*table.Column {
-	if len(t.publicColumns) > 0 {
-		return t.publicColumns
+	if len(t.publicColumns) == 0 {
+		t.publicColumns = t.getCols(full)
```

Comment:
It seems that there is a parallel call problem here, such as multiple sessions using the same table in the same version of Infoschema.